### PR TITLE
test optimization in grouped convolution

### PR DESCRIPTION
**Pull request by affanv14**
There seems to be an optimization error for cudnn. Thanks @gvtulder  for pointing that out

### CODE REVIEW

**Comment by nouiz** [comment]
jenkins test this

**Comment by nouiz** [comment]
jenkins test this

**Comment by nouiz** [comment]
jenkins failed:

http://earlgrey.iro.umontreal.ca:8080/job/Theano_PR/695/testReport/junit/theano.tests/test_flake8/test_format_flake8/

**Comment by affanv14** [comment]
I finished making all the changes

**Comment by nouiz** [comment]
jenkins test this

File: theano/gpuarray/dnn.py [hunk 1]
```diff
@@ -3054,7 +3054,7 @@ def local_abstractconv_cudnn_graph(op, context_name, inputs, outputs):
                         conv_mode=conv_mode,
                         num_groups=op.num_groups)
     elif isinstance(op, AbstractConv2d_gradWeights):
-        shape = (inp2.shape[1], inp1.shape[1],
+        shape = (inp2.shape[1], inp1.shape[1] // op.num_groups,
```
Comment (nouiz) [inline theano/gpuarray/dnn.py]: Can you confirm that the consequence of those bug where crashes?

Comment (affanv14) [inline theano/gpuarray/dnn.py]: yes, this leads to a crash in the cudnn version

Comment (affanv14) [inline theano/gpuarray/dnn.py]: Sorry I missed this

Comment (nouiz) [inline theano/gpuarray/dnn.py]: It happen to everybody. I like to make the distinction between a few catétory of bug depending of how it affect user. Mostly, is it a user visible bug or not.

If it is a crash, it is not the worst bug, as user see it clearly, it is not too bad.

The worst bug are if we return wrong result for some reasons.

**Pull request merged**